eni/attach.go

```diff
@@ -0,0 +1,86 @@
+package eni
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/giantswarm/microerror"
+)
+
+type Config struct {
+	AWSInstanceID string
+	AwsSession    *session.Session
+	DeviceIndex   int64
+	ForceDetach   bool
+	TagKey        string
+	TagValue      string
+}
+
+type Service struct {
```

Comment:
I have some naming request. Can we call this thing  `type ENI struct` and then have methods like `AttachByTag` Instead of `AttachEniByTag`?

I would also prefer to have that stuff in a single package so this is simpler to use.

like `aws/eni.go`, `aws/eni_attach.go` etc.

Then in using code you can do `eni := aws.NewENI()`. That makes whole naming simpler.

Reply:
ack